frappe/query_builder/terms.py

```diff
@@ -51,6 +52,9 @@ def get_sql(
 			value_sql = self.get_value_sql(quote_char=quote_char, **kwargs)
 			sql = param_wrapper.get_sql(param_value=value_sql, **kwargs)
 		else:
+			# * BUG: pypika doesen't parse timedeltas
+			if isinstance(self.value, timedelta):
+				self.value = str(self.value)
```

Comment:
Not really a good idea. hour ranges out of the `0..23` range get stringified to something like `"5 days, 32:02.1211"`

Reply:
There is a nice ISO format for time durations ([ISO8601#Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations)) but `datetime` in python doesn't support it for `timedelta` yet. We can add utils to translate it ourselves or maybe use this package [isodate](https://pypi.org/project/isodate/)

Reply:
Added a format_timedelta util to handle this "better"; Added tests too.